.github/workflows/tests.yaml

```diff
@@ -7,13 +7,13 @@ jobs:
       fail-fast: false
       matrix:
         target:
-        - linux-amd64-fmt
         - linux-amd64-integration-1-cpu
         - linux-amd64-integration-2-cpu
         - linux-amd64-integration-4-cpu
         - linux-amd64-unit-4-cpu-race
-        - all-build
         - linux-386-unit-1-cpu
+        - linux-amd64-e2e
```

Comment:
Do we need to merge the e2e tests ? 

The benefit of keeping them separately is that 'on flake' only a small subset needs to be retried. 

I assume the goal is to limit number of files with test results, but I'm not sure if that pays off.

Reply:
I don't think that non-maintainers can retry tests the same way. They can only push another commit that retriggers all tests.

Goal is to have one workflow that generates junit reports as it makes fetching results easier (one place instead of multiple) and makes it possible to use https://github.com/marketplace/actions/publish-unit-test-results which needs to run once when junit reports are generated.

Reply:
Reverted the change, let's start from unit test and decide later how to handle other test scenarios.